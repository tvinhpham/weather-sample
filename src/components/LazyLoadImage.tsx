import React, { FC, HtmlHTMLAttributes } from 'react';
import Spinner from 'react-bootstrap/Spinner';
import LazyLoad from 'react-lazyload';

const LazyLoadImage: FC<{ src: string; heightRatio?: number } & HtmlHTMLAttributes<HTMLDivElement>> = ({
  src,
  heightRatio = 1,
  style,
  className,
  ...props
}) => {
  const lazyMask = <Spinner animation="grow" className="position-absolute w-100 h-100" />;
  return (
    <div
      className={`w-100 position-relative ${className}`}
      style={{ paddingTop: heightRatio * 100 + '%', overflow: 'hidden', objectFit: 'cover', ...style }}
      {...props}
    >
      <LazyLoad placeholder={lazyMask} once>
        <img src={src} className="d-none" alt="" />
        <div
          className="position-absolute h-100 w-100"
          style={{
            top: 0,
            backgroundSize: 'cover',
            backgroundColor: '#e6e6e6ba',
            backgroundImage: `url(${src})`,
            backgroundRepeat: 'no-repeat',
            backgroundPosition: '50% 50%'
          }}
        />
      </LazyLoad>
    </div>
  );
};

export default LazyLoadImage;
