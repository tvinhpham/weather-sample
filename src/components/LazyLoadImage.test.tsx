import React from 'react';
import renderer from 'react-test-renderer';
import LazyLoadImage from './LazyLoadImage';

it('lazy rendering image', () => {
  const lazyImg = renderer
    .create(<LazyLoadImage src="/logo.svg" heightRatio={0.6}/>)
    .toJSON();
  expect(lazyImg).toMatchSnapshot();
});
