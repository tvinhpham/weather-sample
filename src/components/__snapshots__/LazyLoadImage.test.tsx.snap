// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lazy rendering image 1`] = `
<div
  className="w-100 position-relative undefined"
  style={
    Object {
      "objectFit": "cover",
      "overflow": "hidden",
      "paddingTop": "60%",
    }
  }
>
  <div
    className="lazyload-wrapper "
  >
    <div
      className="position-absolute w-100 h-100 spinner-grow"
    />
  </div>
</div>
`;
